torch/_inductor/scheduler.py

```diff
@@ -1224,9 +1247,42 @@ def __init__(
             for name in other_node.get_names():
                 self.name_to_node[name] = other_node
 
-        self.group = (nodes[0].get_device(), ((sympy.Expr("foreach"),),))
-
+        self.use_custom_partition_algo = use_custom_partition_algo
+        self.group = (snodes[0].get_device(), ((sympy.Expr("combo_kernel"),),))
         self.origins: Set[torch.fx.Node] = set()
+        self.enable_autotune = enable_autotune
+
+    @classmethod
+    def combinable_nodes(
```

Comment:
did not yet review the logic in scheduler. Maybe it would better to do a stack with the automatic horizontal fusion as a second pr. we should also do some enabling by default runs on dashboard to flush out issues. 

I think we will need to be more careful about increasing max memory, or perhaps limit horizontal fusion within more narrow search space.

Reply:
Yes I think it is a good idea to have a second PR to enable automatic horizontal fusion as a second PR. I will work on it once this PR is done.